File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <math.h>

using namespace std;

vector<bool> get_prime_v(int n) {
    vector<bool> primes(n + 1, true);
    primes[0] = false;
    primes[1] = false;
    
    for(int i = 2 ; i <= sqrt(n) ; i++)
        if(primes[i])
            for(int j = i * i ; j <= n ; j+=i)
                primes[j] = false;
    
    
    return primes;
}

int solution(int n) {
    int answer = 0;
    vector<bool> primes = get_prime_v(n);
    
    for(int i = 2 ; i <= n ; i++)
        if(primes[i])
            answer++;
    
    return answer;
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    
    public boolean[] getPrimeArr(int n){
        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = false; isPrime[1] = false;
        
        for(int i = 2 ; i <= (int)Math.sqrt(n) ; i++)
            if(isPrime[i])
                for(int j = i * i ; j <= n ; j += i)
                    isPrime[j] = false;
        
        return isPrime;
        
    }
    
    
    public int solution(int n) {
        boolean[] isPrime = getPrimeArr(n);
        return (int)IntStream.range(1, n + 1)
            .filter(i -> isPrime[i])
            .count();
        
    }
}